text/0000-project-goals.md

```diff
@@ -0,0 +1,196 @@
+* Helping the Foundation and Project to communicate
+    * One challenge for the Rust Foundation has been the lack of clarity around project goals. Programs like fellowships, project grants, etc. have struggled to identify what kind of work would be useful in advancing project direction.
+    * **Declaring goals, and especially goals that are desired but lack owners to drive them, can be very helpful here.**
+* Helping people to get paid for working on Rust
+    * A challenge for people who are looking to work on Rust as part of their job -- whether that be full-time work, part-time work, or contracting -- is that the employer would like to have some confidence that the work will make progress. Too often, people find that they open RFCs or PRs which do not receive review, or which are misaligned with project priorities. A secondary problem is that there can be a perceived conflict-of-interest because people's job performance will be judged on their ability to finish a task, such as stabilizing a language feature, which can lead them to pressure project teams to make progress.
+    * **Having the project agree before-hand that it is a priority to make progress in an area and in particular to aim for achieving particular goals by particular dates will align the incentives and make it easier for people to make commitments to would-be employers.**
+
+For more details, see
+
+* [Blog post on nikomatsakis's blog about project goals](https://smallcultfollowing.com/babysteps/blog/2023/11/28/project-goals/)
+* [Blog post on nikomatsakis's goal about goal ownership](https://smallcultfollowing.com/babysteps/blog/2024/04/05/ownership-in-rust/)
+* [nikomatsakis's slides from the Rust leadership summit](https://github.com/nikomatsakis/team-goals-2024)
+* [Zulip topic in #council stream](https://rust-lang.zulipchat.com/#narrow/stream/392734-council/topic/Project.2Fteam.20goals). This proposal was also discussed at the leadership council meeting on 2024-04-12, during which meeting the council recommended opening an RFC.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## The plan for 2024
+
+The plan is to do a "dry run" of the process in the remainder of 2024. The 2024 process will be driven by nikomatsakis; one of the outputs will be an RFC that proposes a more permanent process for use going forward. The short version of the plan is that we will
+
+* **ASAP (April):** Have a ~2 month period for selecting the initial slate of goals. Goals will be sourced from Rust teams and the broader community. They will cover the highest priority work to be completed in the second half of 2024.
+* **June:** Teams will approve the final slate of goals, making them 'official', and we will author an RFC proposing a refined process.
+* **Remainder of the year:** Regular updates on goal progress will be posted
+* **October:** Presuming all goes well, the process for 2025 H1 begins.
```

Comment:
I think a better time to approve an RFC for the long-term is *after* a set of goals has been decided for the next 6 months.

This might seem a bit weird, but we'll know if its working before then. Having a set of goals essentially is the "proof" though that it was helpful and teams want to continue. So, I view the experimental period defined by this RFC will be through that, but not necessarily extending beyond that.

Also, I would expect there to be overlap in the process for different goal period? Like, October doesn't actually *start* the new goals, only getting them from the teams? Otherwise, October isn't quite 6 months for the initial goals to be worked on.